Aplib.Tests/Core/Belief/BeliefsetTests.cs

```diff
@@ -1,40 +1,143 @@
 ﻿using Aplib.Core.Belief;
 
 namespace Aplib.Tests.Core.Belief;
+
+/// <summary>
+/// Describes a set of tests for the <see cref="BeliefSet"/> class.
+/// </summary>
 public class BeliefSetTests
 {
     /// <summary>
-    /// Given a BeliefSet instance with multiple beliefs,
+    /// Given a BeliefSet instance with multiple <i>public field</i> beliefs,
     /// When UpdateBeliefs is called,
     /// Then all beliefs are updated.
     /// </summary>
     [Fact]
-    public void UpdateBeliefs_Called_UpdatesAllBeliefs()
+    public void UpdateBeliefs_PublicBeliefFields_UpdatesAllBeliefs()
     {
         // Arrange
-        TestBeliefSet beliefSet = new();
+        TestBeliefSetPublic beliefSet = new();
 
         // Act
+        // UpdateBeliefs should set Updated to true for all beliefs.
         beliefSet.UpdateBeliefs();
 
         // Assert
         Assert.True(beliefSet.Belief1.Updated);
         Assert.True(beliefSet.Belief2.Updated);
     }
 
-    private class TestBeliefSet : BeliefSet
+    /// <summary>
+    /// Given a BeliefSet instance with multiple <i>public property</i> beliefs,
+    /// When UpdateBeliefs is called,
+    /// Then all beliefs are updated.
+    /// </summary>
+    [Fact]
+    public void UpdateBeliefs_PublicBeliefProperties_DoesNotUpdateAnyBeliefs()
```

Comment:
Allowing properties to be updated could lead to unexpected behavior very quickly. Consider a belief set as follows:
```cs
private class TestBeliefSet : BeliefSet
{
    private SimpleBelief _belief1 = new();
    
    private SimpleBelief _belief2 = new();
    
    public bool MyBool { get; set; } = true;

    public SimpleBelief MyBelief => MyBool ? _belief1 : _belief2;
}
```
On creation, MyBool is true, so the _private_ _belief1 field is added to the update array and updated on every UpdateBelief() call. Now suppose MyBool is set to false at some point. Then from that point onwards, MyBelief returns _belief2. However, _belief2 is **not** being updated as one might expect. 

There are still some catches with public fields, as they could be assigned to by the user at runtime, but it greatly restricts the number of ways users could mess up.

Reply:
Regarding how the user could mess up with public fields; we could only update _readonly public IBelief fields_. I don't see how the user could mess up then, but it might be less user-friendly having to add the readonly keyword every time.

Reply:
We could filter over private setters. We should thinks this more out later, but this PR has been closed. I was about to resolve all our conversations :(